Model/Schema.php

```diff
@@ -157,21 +165,45 @@ public function prepareOrderData(Order $order)
             }
             $orderData['order_id'] = $order->getIncrementId();
             $orderData['order_date'] = $order->getCreatedAt();
+            $orderData['fulfillment_date'] = $this->getOrderFulfillmentDate($order);
             $orderData['currency_iso'] = $order->getOrderCurrency()->getCode();
             $orderData['email'] = $order->getCustomerEmail();
             $orderData['customer_name'] = trim($order->getCustomerFirstName() . ' ' . $order->getCustomerLastName());
+            $orderData['platform'] = 'magento';
             if (!$orderData['customer_name'] && ($billingAddress = $order->getBillingAddress())) {
                 $orderData['customer_name'] = trim($billingAddress->getFirstname() . ' ' . $billingAddress->getLastname());
             }
             if (!$order->getCustomerIsGuest()) {
                 $orderData['user_reference'] = $order->getCustomerId();
             }
-            $orderData['platform'] = 'magento';
+            if (($fulfillmentDate = $this->getOrderFulfillmentDate($order))) {
+                $orderData['fulfillment_status'] = 'fulfilled';
+                $orderData['fulfillment_status_date'] = $fulfillmentDate;
+            }
         } catch (\Exception $e) {
             $this->yotpoConfig->log("Schema::prepareOrderData() - exception: " . $e->getMessage() . "\n" . $e->getTraceAsString(), "error");
             return [];
         }
 
         return $orderData;
     }
+
+    private function getOrderFulfillmentDate(Order $order)
+    {
+        $lastStatus = $this->orderStatusHistoryFactory->create()->getCollection()
+            ->addFieldToSelect(["id","created_at"])
+            ->addFieldToFilter("order_id", $order->getId())
+            ->addFieldToFilter("store_id", $order->getStoreId())
```

Comment:
will 
 ->addFieldToFilter("store_id", $order->getStoreId())
work for multistore? will it work for single store mode?

Reply:
It's taking the store_id from the order itself, so it should work anyway.